src/compiler/checker.ts

```diff
@@ -17790,11 +17790,11 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 const context = createInferenceContext(freshParams, /*signature*/ undefined, InferenceFlags.None);
                 if (freshMapper) {
                     const freshCombinedMapper = combineTypeMappers(mapper, freshMapper);
-                    for (const p of freshParams) {
-                        if (root.inferTypeParameters.indexOf(p) === -1) {
+                    forEach(freshParams, (p, i) => {

```

Comment:
I'd just prefe `let i = ...` loop to be honest

Reply:
I was going to do that but then I saw that there are more `forEach` uses than `for (let i` in the checker. I don't really care either way.

Not sure this PR even matters, honestly; how many type parameters can one really have? But, it feels clearer to intent that there's nothing fancy going on to do with the order.